From e5a54c19c8455b9343c1dea93d291661c8dca8d8 Mon Sep 17 00:00:00 2001
From: Albin B <albin.bala.krishnan@intel.com>
Date: Wed, 9 Sep 2015 11:52:29 +0530
Subject: [PATCH] [EM]: Remove USB OTG events from typec detect driver

In case of DCP charger, if EM driver sends VBUS event to OTG driver,
then OTG driver is not moving into D3 state. Also there is no USB data
communication in case of DCP charging. So the otg notification is not
sending to otg driver in case of DCP/SE1/ACA charger plug/unplug
events. Hence to send the otg notification based on the cable type,
removing from typec detect driver, which can be send from pmic ccsm
detection driver.

Change-Id: Ie6f3d8dd28d9535651214d2bdb9a551058b01d54
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-18205
Signed-off-by: Albin B <albin.bala.krishnan@intel.com>
---
 drivers/usb/typec/usb_typec_detect.c | 21 ---------------------
 1 file changed, 21 deletions(-)

diff --git a/drivers/usb/typec/usb_typec_detect.c b/drivers/usb/typec/usb_typec_detect.c
index 344769d..98509cf1 100644
--- a/drivers/usb/typec/usb_typec_detect.c
+++ b/drivers/usb/typec/usb_typec_detect.c
@@ -199,8 +199,6 @@ static int typec_detect_send_psy_notification(struct typec_detect *detect,
 static void typec_detect_notify_extcon(struct typec_detect *detect,
 						char *type, bool state)
 {
-	bool notify_otg = false;
-	int otg_evt;
 	enum typec_cable_type cbl_type;
 
 	dev_dbg(detect->phy->dev, "%s: type = %s state = %d\n",
@@ -237,14 +235,6 @@ static void typec_detect_notify_extcon(struct typec_detect *detect,
 			break;
 
 		detect->usb_host_state = state;
-		/* Send ID notification to USB subsystem so that
-		 *  system will switch host mode of operation.
-		 */
-		notify_otg = true;
-		if (state)
-			otg_evt = USB_EVENT_ID;
-		else
-			otg_evt = USB_EVENT_NONE;
 		break;
 
 	case E_TYPEC_CABLE_USB:
@@ -252,14 +242,6 @@ static void typec_detect_notify_extcon(struct typec_detect *detect,
 			break;
 
 		detect->usb_state = state;
-		/* Send ID notification to USB subsystem so that
-		 *  system will switch device mode of operation.
-		 */
-		notify_otg = true;
-		if (state)
-			otg_evt = USB_EVENT_VBUS;
-		else
-			otg_evt = USB_EVENT_NONE;
 		break;
 
 	case E_TYPEC_CABLE_DP_SRC:
@@ -271,9 +253,6 @@ static void typec_detect_notify_extcon(struct typec_detect *detect,
 
 	extcon_set_cable_state(detect->edev, type, state);
 
-	if (notify_otg)
-		atomic_notifier_call_chain(&detect->otg->notifier,
-						otg_evt, NULL);
 notify_ext_err:
 	mutex_unlock(&detect->lock);
 }
-- 
1.9.1

